src/compiler/semantic/analyser.rs

```diff
@@ -250,12 +268,18 @@ impl Analyser {
     fn analyse_statement_recursive(&mut self, stmt: Statement<BigInt, Identifier>) {
         match stmt {
             Statement::Assert(_, expr) => self.analyse_expression(expr),
-            Statement::SignalAssignment(_, _, exprs) => exprs
-                .into_iter()
-                .for_each(|expr| self.analyse_expression(expr)),
-            Statement::SignalAssignmentAssert(_, _, exprs) => exprs
-                .into_iter()
-                .for_each(|expr| self.analyse_expression(expr)),
+            Statement::SignalAssignment(_, ids, exprs) => {
+                self.extract_id_usages(&ids);
```

Comment:
@alxkzmn wouldn't it be better to extract the usages after the semantic analysis? I mean swaping the calls to extract_id_usages and analyse_expression ?

Reply:
I agree, at least it seems more intuitively to make more sense to analyse first and then extract the usages

Reply:
The IDs weren't initially used (_), so shouldn't be any difference (only affects the order in which they are recorded). But I'll rename and move the method after the analysis, just in case.